src/utility/components.js

```diff
@@ -429,55 +429,104 @@ export class Chant {
+   *  <li> The finalis pitch is 'E', suggesting modes 3 and 4. </li>
+   *  <li> Regarding <b> repercussio </b>:
+   *    <ul>
+   *      <li>
+   *        For authentic mode (mode 3): 'C' and 'B' are not among the most repeated notes.
+   *        Therefore, there is a 0% probability of being in authentic mode.
+   *      </li>
+   *      <li>
+   *        For plagal mode (mode 4): 'A' is the second most repeated note (15 times) after a note other than the finalis 'E'.
+   *        Therefore, there is a 50% probability of being in plagal mode.
+   *      </li>
+   *      <li>
+   *        <b> Ambitus suggests plagal mode '4' with 94.87% of notes in range </b>
+   *      </li>
+   *    </ul>
+   *  </li>
+   *  <li> Regarding <b> ambitus </b>:
+   *    <ul>
+   *      <li>
+   *        For the authentic mode (mode 3): 92.31% of the notes are within the range 'E-E'
+   *      </li>
+   *      <li>
+   *        For the plagal mode (mode 4): 94.87% of the notes are within the range 'B-B'
+   *      </li>
+   *      <li>
+   *        <b> Ambitus suggests plagal mode '4' with 94.87% of notes in range </b>
+   *      </li>
+   *    </ul>
+   *  </li>
+   *  <li> <b> Authentic mode '3' has 46.15% rating | Plagal mode '4' has 72.44% raging </b> </li>
+   * </ul>
+   * 
+   * <p><b> Authentic mode '3' has 46.15% rating | Plagal mode '4' has 72.44% raging </b></p>
+   * @returns {[number, number, string]} the mode, the overall rating, and the descripion of the chant.
    */
   calculateSquareMode() {
+    if (env == 'development') {
+      console.log(`~ Square mode of ${this.fileName} ~`);
+    }
     // Combine the 3 conditions to determine the mode
     let mode = -1;    // default undefined mode
     let rating = 0;   // default undefined rating
-    let modeDescription = '';
+
+    let modeDescription = document.createElement('ul');
 
     /** 
-     * 1st condtion (34%): FINALIS - the last note's pitch of the Square notation chant
+     * 1st step: detecting the FINALIS - the last note's pitch of the Square notation chant
      * @type {NeumeComponentSQ}
      */
-    const finalisPitch = this.neumeComponents[this.neumeComponents.length - 1].getPitch();
-    modeDescription += `Finalis pitch is '${finalisPitch}'.\n`;
+    const finalisNC = this.neumeComponents[this.neumeComponents.length - 1];
 
+    const finalisPitch = finalisNC.getPitch();
     let authenticMode = -1, plagalMode = -1;
-    let authenticRepercussio = '', plagalRepercussio = '';
+    let authenticRepercussioPitch = '', plagalRepercussioPitch = '';
 
     if (finalisPitch === 'd') {
-      rating += 0.34;
       authenticMode = 1;
-      authenticRepercussio = 'a';
+      authenticRepercussioPitch = 'a';
       plagalMode = 2;
-      plagalRepercussio = 'f';
+      plagalRepercussioPitch = 'f';
     } else if (finalisPitch === 'e') {
-      rating += 0.34;
       authenticMode = 3;
-      authenticRepercussio = 'c'; // or 'b'
+      authenticRepercussioPitch = 'c'; // or 'b'
```

Comment:
Missing the evaluation of "B" (here only C is being considered).

This can be seen in **Chant 26**:

This shows the frequency of the pitches in this chant = {"f":25, “g":15, “a":7, “e":4, “b":4, “d":2, “c":2}

So for authentic mode (_mode 5_):
- If we consider C to be the repercussio, the percentage would be 0%
- If we consider B as the repercussio, the percentage 12.5% (tied with E pitch too)

Reply:
Same with **Chant 18**:

This shows the frequency of the pitches in this chant = {"g":19, “b":13, “a":21, “c":11, “f":4, “d":6, “e":3}

So for authentic mode (_mode 3_):
- If we consider C as the repercussio, we have 12.5%
- If we consider B as the repercussio, we have 25%